server/handler.go

```diff
@@ -403,11 +424,42 @@ func (h *Handler) AddRemovePeerOperator(regionID uint64, fromStoreID uint64) err
 	return nil
 }
 
+// AddMergeRegionOperator adds an operator to merge region.
+func (h *Handler) AddMergeRegionOperator(regionID uint64, targetID uint64) error {
+	c, err := h.getCoordinator()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	region := c.cluster.GetRegion(regionID)
+	if region == nil {
+		return ErrRegionNotFound(regionID)
+	}
+
+	target := c.cluster.GetRegion(targetID)
+	if target == nil {
+		return ErrRegionNotFound(targetID)
+	}
+
+	if bytes.Compare(region.StartKey, target.EndKey) != 0 && bytes.Compare(region.EndKey, target.StartKey) != 0 {
+		return ErrRegionNotAdjacent
+	}
```

Comment:
What if merge the first region (start key is nil) with the last region (end key is nil) ?